std/sumtype.d

```diff
@@ -714,16 +714,24 @@ public:
     {
         static if (is(This == Rhs))
         {
-            return AliasSeq!(this, rhs).match!((ref value, ref rhsValue) {
-                static if (is(typeof(value) == typeof(rhsValue)))
-                {
-                    return value == rhsValue;
-                }
-                else
+            final switch(tag)
+            {
+                static foreach(idx, T; Types)
                 {
-                    return false;
+                    case idx:
+                        final switch (rhs.tag)
+                        {
+                            static foreach(ridx, U; typeof(rhs).Types)
+                            {
+                                case ridx:
+                                    static if (is(immutable T == immutable U))
```

Comment:
This line introduces a bug. In the original version, the following unit test passes:
```d
@safe unittest
{
    int* p = null;
    const int* cp = null;

    SumType!(int*, const(int*)) x = p;
    SumType!(int*, const(int*)) y = cp;

    assert(x != y);
}
```
With this PR, the unit test above will fail.